drivers/include/dht.h

```diff
@@ -72,6 +88,7 @@ typedef struct {
     dht_params_t params;    /**< Device parameters */
     dht_data_t last_val;    /**< Values of the last measurement */
     uint32_t last_read_us;  /**< Time of the last measurement */
+    uint32_t data_hold_time;/**< Measured minimal data hold time */
```

Comment:
This information can not be gained by using ztimer_usec (may stop at any time if not acquired and it should not be acquired while not in use) data hold also is an bad idea with a sensor that allready reports an old value

I think the whole value caching should not be part of a sensor driver

Reply:
> I think the whole value caching should not be part of a sensor driver

I couldn't disagree stronger. A device driver fundamentally is an hardware abstraction layer that hides the nitty-gritty low level hardware details from the users and should make it easy to use the hardware.

It is absolutely unrealistic to assume that all user check the datasheet of every piece of hardware they use first for nitty gritty details like a max polling rate. Hence, if the driver would have a "yolo" attitude to the max polling rate it will damage the equipment of some users. IMO they would be rightfully angry with us not even trying to prevent that damage.

I personally expect that drivers behave sane and safe by default and I'm sure many others also have the same expectation.

> data hold also is an bad idea with a sensor that allready reports an old value

This is a room temperature and room humidity sensor and not some industrial process monitoring sensor. The impact of the caching is negligible to the time resolution, but it prevents hardware damage.

> may stop at any time if not acquired

A `ztimer_acquire` on `ZTIMER_USEC` would indeed not be ideal. Better would be setting a timer that clears a "cached value is still good" flag once the max polling period has passed. That could even be done on a `ZTIMER_MSEC` clock if `MODULE_ZTIMER_MSEC` is used anyway to still allow entering low power modes while the timer is set.

Reply:
Let's not get this PR stuck on this discussion. I'm fine with whatever outcome in this regard.

[Someone testing with 0.35 period readings](https://www.kandrsmith.org/RJS/Misc/Hygrometers/dht_sht_how_fast.html) ended up heating up the sensor by only 1 ℃. That doesn't sound like something to worry about other than messing with the accuracy of the readings.

Reply:
I have read several of these horribly translated datasheets and have never seen any mention of a 5-second interval. What I have understood and can easily observe with a 5-second interval is that the values returned by the sensor are those measured at the time of the previous reading. For example, if we take a reading at t1 and another at t2, the values returned at t2 are those that were measured at t1:
```
2023-06-14 01:05:16,959 # DHT values - temp: 25.3°C - relative humidity: 68.9%
2023-06-14 01:05:21,986 # DHT values - temp: 25.3°C - relative humidity: 68.9%
2023-06-14 01:05:27,012 # DHT values - temp: 25.3°C - relative humidity: 68.9%
2023-06-14 01:05:32,038 # DHT values - temp: 25.3°C - relative humidity: 68.9%
2023-06-14 01:05:37,064 # DHT values - temp: 25.3°C - relative humidity: 68.9%
```
I put my finger on the sensor for just one second at 05:38.
```
2023-06-14 01:05:42,090 # DHT values - temp: 25.3°C - relative humidity: 68.9%
2023-06-14 01:05:47,117 # DHT values - temp: 25.5°C - relative humidity: 72.9%
2023-06-14 01:05:52,143 # DHT values - temp: 25.4°C - relative humidity: 70.9%
2023-06-14 01:05:57,170 # DHT values - temp: 25.4°C - relative humidity: 69.6%
2023-06-14 01:06:02,196 # DHT values - temp: 25.4°C - relative humidity: 68.8%
```
The data caching is not my idea; it was already present in the previous driver with a constant DATA_HOLD_TIME of one second. I don't mind removing it as long as the driver does not prevent me from reading my sensors twice per second.

Reply:
> Let's not get this PR stuck on this discussion. I'm fine with whatever outcome in this regard.

May I re-squash now ?

Reply:
> Citing maribu:  please squash